src/libaktualizr/uptane/uptane_test.cc

```diff
@@ -918,6 +1090,36 @@ TEST(Uptane, offlineIteration) {
   EXPECT_TRUE(sota_client->uptaneOfflineIteration(&targets_offline, nullptr));
   EXPECT_EQ(targets_online, targets_offline);
 }
+/*
+ Ignore updates for unrecognized ECUs.
+*/
+TEST(Uptane, DISABLED_IgnoreUnknownUpdate) {
+  // FIXME This test is disabled because it conflicts with some tests in uptane_vector_tests.cc
```

Comment:
What's the problem here? I don't understand how it conflicts.

Reply:
Oh yes, I forgot to tell you about this test. uptane_vector_tests expects exception if we get unknown update. But this test checks, do we ignore unknown update. So we should decide, should we throw exception or ignore.

Reply:
I don't see the conflict. In `SotaUptaneClient::getNewTargets`, it looks to me like unrecognized serials or HW IDs set `last_exception` but then return false. `uptane_vector_tests` just checks for the expected value in `last_exception`. I don't like this whole `last_exception` business, but it isn't a problem here.

Reply:
I am not sure it is good idea. Returning false, means some error happens and we won't be able to install good update, so that's mean we dont ignore bad updates. So we should return true even when wrong ecu update happens.

Reply:
In my opinion, if we are going to ignore updates with wrong ecu, we should enable this test and remove appropriate tests in uptane_vector_tests, or just change them to check if we really ignore bad ecu updates. But it is not so trivial, so my suggestion is to remove them.

Reply:
I have enabled this test with some fixes.